wled00/json.cpp

```diff
@@ -18,6 +18,29 @@ void deserializeSegment(JsonObject elem, byte it, byte presetId)
   uint16_t start = elem[F("start")] | seg.start;
   int stop = elem["stop"] | -1;
 
+    if (elem["n"]) {
+      // name field exists
+      String name = elem["n"];
+      if (name.length()) {
+        if (seg.name) delete seg.name;
+        seg.name = new char[name.length()+1];
```

Comment:
This is not very safe. A long string could be used to run into low memory situations deliberately, or `strcpy` could write to memory if the allocation failed.
We should
- limit length with strlcpy to 32 or at most 64 characters
- check that `seg.name` is not nullptr before copying

Reply:
Thanks for pointing it out. Perhaps delete[] is also in order...